central/complianceoperator/v2/scanconfigurations/service/service_impl.go

```diff
@@ -392,6 +393,53 @@ func (s *serviceImpl) GetMyReportHistory(ctx context.Context, request *v2.Compli
 	return res, nil
 }
 
+func (s *serviceImpl) DeleteReport(ctx context.Context, req *v2.ResourceByID) (*v2.Empty, error) {
+	if req.GetId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Report Snapshot ID is required for deletion")
+	}
+
+	slimUser := authn.UserFromContext(ctx)
+	if slimUser == nil {
+		return nil, errors.New("Could not determine user identity from provided context")
+	}
+
+	snapshot, found, err := s.snapshotDS.GetSnapshot(ctx, req.GetId())
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to retrieve Report Snapshot %s", req.GetId())
+	}
+	if !found {
+		return nil, errors.Wrapf(errox.NotFound, "Unable to find the Report Snapshots %s", req.GetId())
+	}
+
+	if slimUser.GetId() != snapshot.GetUser().GetId() {
+		return nil, errors.Errorf("The user %s cannot delete the report %s", slimUser.GetId(), snapshot.GetReportId())
+	}
+
+	status := snapshot.GetReportStatus()
+	if status.GetReportNotificationMethod() != storage.ComplianceOperatorReportStatus_DOWNLOAD {
+		return nil, errors.Wrapf(errox.InvalidArgs, "The Report Snapshot %s did not generate a downloadable report", req.GetId())
+	}
```

Comment:
So does that mean that we cannot delete it? Can we delete only reports that are meant for downloads?
If yes, then I would suggest:
```suggestion
	if status.GetReportNotificationMethod() != storage.ComplianceOperatorReportStatus_DOWNLOAD {
		return nil, errors.Wrapf(errox.InvalidArgs, "The report %s is not downloadable and cannot be deleted", req.GetId())
	}
```

Reply:
Done